upcloud/service/service.go

```diff
@@ -1,107 +1,98 @@
 package service
 
 import (
+	"context"
 	"encoding/json"
 	"fmt"
+	"net/http"
 
-	"github.com/UpCloudLtd/upcloud-go-api/v4/upcloud"
-	"github.com/UpCloudLtd/upcloud-go-api/v4/upcloud/client"
+	"github.com/UpCloudLtd/upcloud-go-api/v5/upcloud"
+	"github.com/UpCloudLtd/upcloud-go-api/v5/upcloud/client"
 )
 
-type requestable interface {
-	RequestURL() string
-}
-
-type Zones interface {
-	GetZones() (*upcloud.Zones, error)
-}
-
-var _ Zones = (*Service)(nil)
-
-type PriceZones interface {
-	GetPriceZones() (*upcloud.PriceZones, error)
-}
-
-var _ PriceZones = (*Service)(nil)
-
-type TimeZones interface {
-	GetTimeZones() (*upcloud.TimeZones, error)
+type Client interface {
+	// Get performs a GET request to the specified path and returns the response body.
+	Get(ctx context.Context, path string) ([]byte, error)
+	// Post performs a POST request to the specified path and returns the response body.
+	Post(ctx context.Context, path string, body []byte) ([]byte, error)
+	// Put performs a PUT request to the specified path and returns the response body.
+	Put(ctx context.Context, path string, body []byte) ([]byte, error)
+	// Patch performs a PATCH request to the specified path and returns the response body.
+	Patch(ctx context.Context, path string, body []byte) ([]byte, error)
+	// Delete performs a DELETE request to the specified path and returns the response body.
+	Delete(ctx context.Context, path string) ([]byte, error)
+	// Do performs a HTTP request using custom request object and returns the response body.
+	Do(r *http.Request) ([]byte, error)
 }
 
-var _ TimeZones = (*Service)(nil)
-
-type Plans interface {
-	GetPlans() (*upcloud.Plans, error)
+type requestable interface {
```

Comment:
What about calling this `request`?

Reply:
I thought also first but I think that then we would need to alias imports from `request` package. We have had this interface for a while and I struggled with the name already then :)  Current name is not perfect but it seems that it is used else where also to name this kind of interface. Anyway it's private and if we come up with better name (other than request) we can change it later.